src/components/Menu.jsx

```diff
@@ -12,6 +12,8 @@ const propTypes = {
   renderItemLabel: PropTypes.func,
   value: PropTypes.string,
   onSelect: PropTypes.func,
+  shouldTransferFocus: PropTypes.bool,
+  notifyBeforeFocusWrap: PropTypes.func,
 };
```

Comment:
Ah, what I meant for onBeforeItemFocusChange(focusedIndex, nextFocusedIndex, event.key) is that nextFocusedIndex would already include wrapping calculation. So let's say there are 3 items in the list, and the down key is pressed on the last item. It would call onBeforeItemFocusChange(2, 0, 'ARROW_DOWN') -- then there's no need to duplicate any of the default wrapping algorithm in the callback. Menu is just saying, I'm about to do this focus change, should I continue, or do you want to do something else?

In the handler, you can just look at those arguments and determine if it's wrapping, and if so, return null and do the focus handoff.